### PR TITLE
Add the helm executor common package

### PR DESCRIPTION
**Pull request by iawia002**
### What type of PR is this?

/kind feature

### What this PR does / why we need it:

The PR proposes to create a new public repository for packages that will be used by multiple kubesphere projects and add a helm executor package that has almost the same logic as the previous [helm wrapper](https://github.com/kubesphere/kubesphere/blob/master/pkg/simple/client/openpitrix/helmwrapper/helm_wrapper.go), except that it will use a separate job to perform the install/uninstall tasks, the interface is defined as follows:

```go
// Executor is used to manage a helm release, you can install/uninstall and upgrade a chart
// or get the status and manifest data of the release, etc.
type Executor interface {
	// Install installs the specified chart and returns the name of the Job that executed the task.
	Install(ctx context.Context, chartName, chartData, values string) (string, error)
	// Upgrade upgrades the specified chart and returns the name of the Job that executed the task.
	Upgrade(ctx context.Context, chartName, chartData, values string) (string, error)
	// Uninstall is used to uninstall the specified chart and returns the name of the Job that executed the task.
	Uninstall(ctx context.Context) (string, error)
	// Manifest returns the manifest data for this release.
	Manifest() (string, error)
	// IsReleaseReady checks if the helm release is ready.
	IsReleaseReady(timeout time.Duration) (bool, error)
}
```

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
none
```

### Additional documentation, usage docs, etc.:

```docs

```

/cc @wansir @zheng1

### CODE REVIEW

File: staging/src/kubesphere.io/utils/helm/executor.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	// Install installs the specified chart and returns the name of the Job that executed the task.
+	Install(chartName, chartData, values string) (string, error)
+	// Upgrade upgrades the specified chart and returns the name of the Job that executed the task.
+	Upgrade(chartName, chartData, values string) (string, error)
+	// Uninstall is used to uninstall the specified chart and returns the name of the Job that executed the task.
+	Uninstall() (string, error)
+	// Manifest returns the manifest data for this release.
+	Manifest() (string, error)
+	// IsReleaseReady checks if the helm release is ready.
+	IsReleaseReady(timeout time.Duration) (bool, error)
```
Comment (wansir) [inline staging/src/kubesphere.io/utils/helm/executor.go:35]:
```suggestion
	// Install installs the specified chart and returns the name of the Job that executed the task.
	Install(ctx context.Context, chartName, chartData, values string) (string, error)
	// Upgrade upgrades the specified chart and returns the name of the Job that executed the task.
	Upgrade(ctx context.Context, chartName, chartData, values string) (string, error)
	// Uninstall is used to uninstall the specified chart and returns the name of the Job that executed the task.
	Uninstall(ctx context.Context) (string, error)
	// Manifest returns the manifest data for this release.
	Manifest() (string, error)
	// IsReleaseReady checks if the helm release is ready.
	IsReleaseReady(timeout time.Duration) (bool, error)
```

Comment (iawia002) [reply staging/src/kubesphere.io/utils/helm/executor.go]: Updated.

**Pull request merged**